### PR TITLE
Add batch cancellation

### PR DESCRIPTION
**Pull request by romasku**
Closes https://github.com/neuromation/neuro-flow/issues/68

### CODE REVIEW

**Review by asvetlov — CHANGES_REQUESTED** [review]
Looks good, a few minor fixes are required

File: neuro_flow/storage.py [hunk 1]
```diff
@@ -576,6 +585,30 @@ async def finish_batch_action(
...
+        now = datetime.datetime.now(datetime.timezone.utc)
+        ret = FinishedTask(
+            attempt=attempt,
+            id=task.id,
+            raw_id="",
+            when=now,
+            status=JobStatus.CANCELLED,
+            exit_code=None,
+            created_at=task.created_at,
+            started_at=task.created_at,
```
[the remark this answers is not in the document]
Comment (romasku) [reply neuro_flow/storage.py:603]: `StartedTask` has no such field. This code is same as in `BatchFSStorage.finish_batch_action`.

[the remark this answers is not in the document]
Comment (romasku) [reply neuro_flow/storage.py]: I removed this method at all.

File: neuro_flow/storage.py [hunk 2]
```diff
@@ -576,6 +585,30 @@ async def finish_batch_action(
...
+            finished_at=now,
```
[the remark this answers is not in the document]
Comment (romasku) [reply neuro_flow/storage.py]: I removed this method at all.

File: neuro_flow/batch_runner.py [hunk 3]
```diff
@@ -175,6 +179,24 @@ async def process(
 
                 await self._process_started(attempt, topos, started, finished, skipped)
 
+                # Check for cancellation
+                attempt = await self._storage.find_attempt(bake, attempt.number)
+                if attempt.result == JobStatus.CANCELLED:
+                    for st in started.values():
+                        if st.raw_id:
+                            await self._client.jobs.kill(st.raw_id)
```
[the remark this answers is not in the document]
Comment (romasku) [reply neuro_flow/batch_runner.py:188]: Thanks, that was good idea and it even simplified the code!

**Review by asvetlov — APPROVED** [review]
lgtm

**Pull request merged**